src/shogun/lib/observers/ObservedValue.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Giovanni De Toni
+ */
+
+#ifndef SHOGUN_OBSERVEDVALUE_H
+#define SHOGUN_OBSERVEDVALUE_H
+
+#include <shogun/base/SGObject.h>
+
+namespace shogun {
+
+	template<class T>
+	class ObservedValueTemplated;
+
+	/**
+ 	* Observed value which is emitted by algorithms.
+ 	*/
+	class ObservedValue : public CSGObject {
+	public:
+		/**
+		 * Constructor
+		 * @param step step
+		 * @param name name of the observed value
+		 */
+		ObservedValue(const int64_t step, const std::string &name);
+
+		/**
+		 * Destructor
+		 */
+		~ObservedValue() {};
+
+#ifndef SWIG
+
+		/**
+		* Return a any version of the stored type.
+		* @return the any value.
+		*/
+		virtual Any get_any() const {
```

Comment:
maybe `const Any&` instead to avoid a copy?

Reply:
+1